### ISSUE TITLE
AssertionError from scripts_merge() in bash.py

### ISSUE THREAD

**Issue by msc51** [PacificBiosciences/FALCON#505]
Dear Falcon developers,

I am at the "merge-scatter" step and encountered an AssertionError in line 310 of bash.y (function "scripts_merge"):
assert las_fn == os.path.basename(las_files[0])

I printed the values of some variables and I saw that:
db_prefix = raw_reads
p_id = 1
las_fn = raw_reads.1.las
las_files = ['L2.1.1.las', 'L1.1.1.las', 'L1.1.2.las', 'L1.1.3.las', ...]

Since las_fn is "raw_reads.1.las" and las_files[0] is "L2.1.1.las", the assertion statement won't pass. I want to report this as I think this may be a bug. 

I just turned the assertion statement into an assignment:
 las_fn == os.path.basename(las_files[0])
and the L2 merges are underway...

**Comment by pb-cdunn** [comment]
* https://github.com/PacificBiosciences/FALCON/blob/master/falcon_kit/bash.py#L310

Yes, this is a bug. This is the first time I've seen L2 files. Really, we need to replace HPC.daligner since it does different things for different numbers of merge levels. We are constantly reverse-engineering it.

Could you attach the output of `HPC.daligner`? I'm at home right now, without a run-dir, so I don't know exactly what you can look for it in your directory tree.

Yes, your assignment should be ok at first, but I'm not sure what happens later. Those L2 files will need to be merged into raw_reads files.

**Comment by kevlim83** [comment]
Hi,

Sorry to crash on this thread. But I also encountered the same problem. I tried to debug the same way msc51 mentioned and saw L2.1.1.las too. There were no files with L2 prefix in the 0-rawreads directory.

This is strange because I tried two runs with different parameters (and also different cluster, first was pbs and the second was sge) and first run (on pbs) finished step 0 but I saw this assertion error only in the second run (on sge). (pb-cdunn: Not sure if this additional information is useful for your debugging)

And to confirm, can we just turn the assertion into assignment as a work around?

**Comment by pb-cdunn** [comment]
There could be a bug when we have 2 levels of merging (L2). Could someone show us an HPC.daligner output which contains L2?

**Comment by kevlim83** [comment]
There are no L2 outputs in my job* directories as far as I can tell.

**Comment by pb-cdunn** [comment]
I need to see the file generated by `HPC.daligner`. (I think that's in `0-rawreads/run_jobs.sh`.) Those are the bash instructions on how to merge.

**Comment by msc51** [comment]
Hi Chris,

Sorry for not replying earlier. I went through the whole process successfully so I wasn't even looking back. :) But for debugging the software I thought I should help. 

My run_jobs.sh file is huge and I am not attaching it here. Instead i am attaching an rp_00001.sh file in my 0-rawreads/m_00001/ dir where L2 merging happened. I have always had L2 las files in m_xxxxx, even on previous releases of Falcon. 

[rp_00001.sh.txt](https://github.com/PacificBiosciences/FALCON/files/686677/rp_00001.sh.txt)

**Comment by pb-cdunn** [comment]
It would be helpful if you could post the portion of run_jobs.sh where L2 is first mentioned.

I think our problem is that we attempt to run all merge jobs in parallel, but L2 jobs need to run *after* L1 jobs. That will take some effort.

**Comment by msc51** [comment]
# Level 1 merge jobs (712)
LAmerge -v L2.1.1 L1.1.1 L1.1.2 L1.1.3 L1.1.4 L1.1.5 L1.1.6 L1.1.7 L1.1.8 L1.1.9 L1.1.10 L1.1.11 L1.1.12 L1.1.13 L1.1.14 L1.1.15 L1.1.16 L1.1.17 L1.1.18 L1.1.19 L1.1.20 L1.1.21 L1.1.22 L1.1.23 L1.1.24 L1.1.25 L1.1.26 L1.1.27 L1.1.28 L1.1.29 L1.1.30 L1.1.31 L1.1.32 L1.1.33 L1.1.34 L1.1.35 L1.1.36 L1.1.37 L1.1.38 L1.1.39 L1.1.40 L1.1.41 L1.1.42 L1.1.43 L1.1.44 L1.1.45 L1.1.46 L1.1.47 L1.1.48 L1.1.49 L1.1.50 L1.1.51 L1.1.52 L1.1.53 L1.1.54 L1.1.55 L1.1.56 L1.1.57 L1.1.58 L1.1.59 L1.1.60 L1.1.61 L1.1.62 L1.1.63 L1.1.64 L1.1.65 L1.1.66 L1.1.67 L1.1.68 L1.1.69 L1.1.70 L1.1.71 L1.1.72 L1.1.73 L1.1.74 L1.1.75 L1.1.76 L1.1.77 L1.1.78 L1.1.79 L1.1.80 L1.1.81 L1.1.82 L1.1.83 L1.1.84 L1.1.85 L1.1.86 L1.1.87 L1.1.88 L1.1.89 L1.1.90 L1.1.91 L1.1.92 L1.1.93 L1.1.94 L1.1.95 L1.1.96 L1.1.97 L1.1.98 L1.1.99 L1.1.100 L1.1.101 L1.1.102 L1.1.103 L1.1.104 L1.1.105 L1.1.106 L1.1.107 L1.1.108 L1.1.109 L1.1.110 L1.1.111 L1.1.112 L1.1.113 L1.1.114 L1.1.115 L1.1.116 L1.1.117 L1.1.118 L1.1.119 L1.1.120 L1.1.121 L1.1.122 L1.1.123 L1.1.124 L1.1.125 L1.1.126 L1.1.127 L1.1.128 L1.1.129 L1.1.130 L1.1.131 L1.1.132 L1.1.133 L1.1.134 L1.1.135 L1.1.136 L1.1.137 L1.1.138 L1.1.139 L1.1.140 L1.1.141 L1.1.142 L1.1.143 L1.1.144 L1.1.145 L1.1.146 L1.1.147 L1.1.148 L1.1.149 L1.1.150 L1.1.151 L1.1.152 L1.1.153 L1.1.154 L1.1.155 L1.1.156 L1.1.157 L1.1.158 L1.1.159 L1.1.160 L1.1.161 L1.1.162 L1.1.163 L1.1.164 L1.1.165 L1.1.166 L1.1.167 L1.1.168 L1.1.169 L1.1.170 L1.1.171 L1.1.172 L1.1.173 L1.1.174 L1.1.175 L1.1.176 L1.1.177 L1.1.178
LAmerge -v L2.1.2 L1.1.179 L1.1.180 L1.1.181 L1.1.182 L1.1.183 L1.1.184 L1.1.185 L1.1.186 L1.1.187 L1.1.188 L1.1.189 L1.1.190 L1.1.191 L1.1.192 L1.1.193 L1.1.194 L1.1.195 L1.1.196 L1.1.197 L1.1.198 L1.1.199 L1.1.200 L1.1.201 L1.1.202 L1.1.203 L1.1.204 L1.1.205 L1.1.206 L1.1.207 L1.1.208 L1.1.209 L1.1.210 L1.1.211 L1.1.212 L1.1.213 L1.1.214 L1.1.215 L1.1.216 L1.1.217 L1.1.218 L1.1.219 L1.1.220 L1.1.221 L1.1.222 L1.1.223 L1.1.224 L1.1.225 L1.1.226 L1.1.227 L1.1.228 L1.1.229 L1.1.230 L1.1.231 L1.1.232 L1.1.233 L1.1.234 L1.1.235 L1.1.236 L1.1.237 L1.1.238 L1.1.239 L1.1.240 L1.1.241 L1.1.242 L1.1.243 L1.1.244 L1.1.245 L1.1.246 L1.1.247 L1.1.248 L1.1.249 L1.1.250 L1.1.251 L1.1.252 L1.1.253 L1.1.254 L1.1.255 L1.1.256 L1.1.257 L1.1.258 L1.1.259 L1.1.260 L1.1.261 L1.1.262 L1.1.263 L1.1.264 L1.1.265 L1.1.266 L1.1.267 L1.1.268 L1.1.269 L1.1.270 L1.1.271 L1.1.272 L1.1.273 L1.1.274 L1.1.275 L1.1.276 L1.1.277 L1.1.278 L1.1.279 L1.1.280 L1.1.281 L1.1.282 L1.1.283 L1.1.284 L1.1.285 L1.1.286 L1.1.287 L1.1.288 L1.1.289 L1.1.290 L1.1.291 L1.1.292 L1.1.293 L1.1.294 L1.1.295 L1.1.296 L1.1.297 L1.1.298 L1.1.299 L1.1.300 L1.1.301 L1.1.302 L1.1.303 L1.1.304 L1.1.305 L1.1.306 L1.1.307 L1.1.308 L1.1.309 L1.1.310 L1.1.311 L1.1.312 L1.1.313 L1.1.314 L1.1.315 L1.1.316 L1.1.317 L1.1.318 L1.1.319 L1.1.320 L1.1.321 L1.1.322 L1.1.323 L1.1.324 L1.1.325 L1.1.326 L1.1.327 L1.1.328 L1.1.329 L1.1.330 L1.1.331 L1.1.332 L1.1.333 L1.1.334 L1.1.335 L1.1.336 L1.1.337 L1.1.338 L1.1.339 L1.1.340 L1.1.341 L1.1.342 L1.1.343 L1.1.344 L1.1.345 L1.1.346 L1.1.347 L1.1.348 L1.1.349 L1.1.350 L1.1.351 L1.1.352 L1.1.353 L1.1.354 L1.1.355 L1.1.356
LAmerge -v L2.2.1 L1.2.1 L1.2.2 L1.2.3 L1.2.4 L1.2.5 L1.2.6 L1.2.7 L1.2.8 L1.2.9 L1.2.10 L1.2.11 L1.2.12 L1.2.13 L1.2.14 L1.2.15 L1.2.16 L1.2.17 L1.2.18 L1.2.19 L1.2.20 L1.2.21 L1.2.22 L1.2.23 L1.2.24 L1.2.25 L1.2.26 L1.2.27 L1.2.28 L1.2.29 L1.2.30 L1.2.31 L1.2.32 L1.2.33 L1.2.34 L1.2.35 L1.2.36 L1.2.37 L1.2.38 L1.2.39 L1.2.40 L1.2.41 L1.2.42 L1.2.43 L1.2.44 L1.2.45 L1.2.46 L1.2.47 L1.2.48 L1.2.49 L1.2.50 L1.2.51 L1.2.52 L1.2.53 L1.2.54 L1.2.55 L1.2.56 L1.2.57 L1.2.58 L1.2.59 L1.2.60 L1.2.61 L1.2.62 L1.2.63 L1.2.64 L1.2.65 L1.2.66 L1.2.67 L1.2.68 L1.2.69 L1.2.70 L1.2.71 L1.2.72 L1.2.73 L1.2.74 L1.2.75 L1.2.76 L1.2.77 L1.2.78 L1.2.79 L1.2.80 L1.2.81 L1.2.82 L1.2.83 L1.2.84 L1.2.85 L1.2.86 L1.2.87 L1.2.88 L1.2.89 L1.2.90 L1.2.91 L1.2.92 L1.2.93 L1.2.94 L1.2.95 L1.2.96 L1.2.97 L1.2.98 L1.2.99 L1.2.100 L1.2.101 L1.2.102 L1.2.103 L1.2.104 L1.2.105 L1.2.106 L1.2.107 L1.2.108 L1.2.109 L1.2.110 L1.2.111 L1.2.112 L1.2.113 L1.2.114 L1.2.115 L1.2.116 L1.2.117 L1.2.118 L1.2.119 L1.2.120 L1.2.121 L1.2.122 L1.2.123 L1.2.124 L1.2.125 L1.2.126 L1.2.127 L1.2.128 L1.2.129 L1.2.130 L1.2.131 L1.2.132 L1.2.133 L1.2.134 L1.2.135 L1.2.136 L1.2.137 L1.2.138 L1.2.139 L1.2.140 L1.2.141 L1.2.142 L1.2.143 L1.2.144 L1.2.145 L1.2.146 L1.2.147 L1.2.148 L1.2.149 L1.2.150 L1.2.151 L1.2.152 L1.2.153 L1.2.154 L1.2.155 L1.2.156 L1.2.157 L1.2.158 L1.2.159 L1.2.160 L1.2.161 L1.2.162 L1.2.163 L1.2.164 L1.2.165 L1.2.166 L1.2.167 L1.2.168 L1.2.169 L1.2.170 L1.2.171 L1.2.172 L1.2.173 L1.2.174 L1.2.175 L1.2.176 L1.2.177 L1.2.178
LAmerge -v L2.2.2 L1.2.179 L1.2.180 L1.2.181 L1.2.182 L1.2.183 L1.2.184 L1.2.185 L1.2.186 L1.2.187 L1.2.188 L1.2.189 L1.2.190 L1.2.191 L1.2.192 L1.2.193 L1.2.194 L1.2.195 L1.2.196 L1.2.197 L1.2.198 L1.2.199 L1.2.200 L1.2.201 L1.2.202 L1.2.203 L1.2.204 L1.2.205 L1.2.206 L1.2.207 L1.2.208 L1.2.209 L1.2.210 L1.2.211 L1.2.212 L1.2.213 L1.2.214 L1.2.215 L1.2.216 L1.2.217 L1.2.218 L1.2.219 L1.2.220 L1.2.221 L1.2.222 L1.2.223 L1.2.224 L1.2.225 L1.2.226 L1.2.227 L1.2.228 L1.2.229 L1.2.230 L1.2.231 L1.2.232 L1.2.233 L1.2.234 L1.2.235 L1.2.236 L1.2.237 L1.2.238 L1.2.239 L1.2.240 L1.2.241 L1.2.242 L1.2.243 L1.2.244 L1.2.245 L1.2.246 L1.2.247 L1.2.248 L1.2.249 L1.2.250 L1.2.251 L1.2.252 L1.2.253 L1.2.254 L1.2.255 L1.2.256 L1.2.257 L1.2.258 L1.2.259 L1.2.260 L1.2.261 L1.2.262 L1.2.263 L1.2.264 L1.2.265 L1.2.266 L1.2.267 L1.2.268 L1.2.269 L1.2.270 L1.2.271 L1.2.272 L1.2.273 L1.2.274 L1.2.275 L1.2.276 L1.2.277 L1.2.278 L1.2.279 L1.2.280 L1.2.281 L1.2.282 L1.2.283 L1.2.284 L1.2.285 L1.2.286 L1.2.287 L1.2.288 L1.2.289 L1.2.290 L1.2.291 L1.2.292 L1.2.293 L1.2.294 L1.2.295 L1.2.296 L1.2.297 L1.2.298 L1.2.299 L1.2.300 L1.2.301 L1.2.302 L1.2.303 L1.2.304 L1.2.305 L1.2.306 L1.2.307 L1.2.308 L1.2.309 L1.2.310 L1.2.311 L1.2.312 L1.2.313 L1.2.314 L1.2.315 L1.2.316 L1.2.317 L1.2.318 L1.2.319 L1.2.320 L1.2.321 L1.2.322 L1.2.323 L1.2.324 L1.2.325 L1.2.326 L1.2.327 L1.2.328 L1.2.329 L1.2.330 L1.2.331 L1.2.332 L1.2.333 L1.2.334 L1.2.335 L1.2.336 L1.2.337 L1.2.338 L1.2.339 L1.2.340 L1.2.341 L1.2.342 L1.2.343 L1.2.344 L1.2.345 L1.2.346 L1.2.347 L1.2.348 L1.2.349 L1.2.350 L1.2.351 L1.2.352 L1.2.353 L1.2.354 L1.2.355 L1.2.356
